### ISSUE TITLE
Private calls

### ISSUE THREAD

**Issue by afeno** [MarshalX/tgcalls#176]
Hello,
In the TODO list, it says that "Incoming and Outgoing private calls (already there and working, but not in the release version).".
How can I test this functionality that is already there but not in the release yet?
Thnaks.

**Comment by rojvv** [comment]
You can test this functionality by waiting until it gets released.

**Comment by pandatest2023** [comment]
When will it be released?

**Comment by pringtest** [comment]
GG

**Comment by rojvv** [comment]
@phantom0000 You can know when it gets released by waiting until it gets released.

**Comment by tgtglove** [comment]
> Hello,
> In the TODO list, it says that "Incoming and Outgoing private calls (already there and working, but not in the release version).".
> How can I test this functionality that is already there but not in the release yet?
> Thnaks.

Now I have a solution.

**Comment by Day0Dreamer** [comment]
> You can test this functionality by waiting until it gets released.

This test fails. It is a bad advice.

**Comment by moha-abdi** [comment]
you can just use the raw api if you cant wait its not that hard to make encrypted calls reffer to the telegram docs

**Comment by afeno** [comment]
Do you know how to do that? Because many people tried without success...

**Comment by moha-abdi** [comment]
> 

Yeah I already made one using pyrogram and raw functions just follow instructions to make end to end encryption that telegram stated the rest is not that hard

**Comment by afeno** [comment]
Are you able to play an audio file on both Android and iPhone Telegram clients? Or just to start the call (Ring the phone).

**Comment by moha-abdi** [comment]
I can play audio, also you can check tests.py file in thcalls it has both incoming and outgoing calls

**Comment by afeno** [comment]
If you get pytgcalls working with private calls (ring the phone and play the audio) I can sponsor or donate some money.

**Comment by georgerabus** [comment]
> > 
> 
> Yeah I already made one using pyrogram and raw functions just follow instructions to make end to end encryption that telegram stated the rest is not that hard

can you please show me how you did it? I had some issues like RPCError 400: CALL_PROTOCOL_LAYER_INVALID

**Comment by greymattar** [comment]
> > 
> 
> Yeah I already made one using pyrogram and raw functions just follow instructions to make end to end encryption that telegram stated the rest is not that hard

Did u do it using pytgcalls ? because i can't seem to import the necessary functions and classes to use the group call features .
